perf/pgbench.py

```
@@ -0,0 +1,127 @@
+            self.cancel("This script requires root privileges, "
+                        "Please try again with 'sudo' or as 'root'")
+
+        # check if pgbench is installed
+        if (self.sm.check_installed("pgbench")):
+            self.log.info("Pgbench is already installed")
+        else:
+            pkg_mgr = self.get_pkg_manager()
+            if pkg_mgr:
+                self.install_packages(pkg_mgr)
+            else:
+                self.cancel("Unsupported Linux distribution")
+
+        # add current user to db
+        output = self.run_cmd(f"sudo -u postgres createuser --superuser {os.getlogin()}", True)
+        if output[0] and "already exists" not in output[1]:
+            self.cancel(f"Could not create role {os.getlogin()}")
+
+    def test(self):
+
+        # Setup the params
+        scaling_factor = self.params.get("scaling_factor", default=100)
+        worker_threads = self.params.get("worker_threads", default=cpu.online_count())
+        db_clients = self.params.get("db_clients", default=cpu.online_count())
+        protocol = self.params.get("protocol", default="prepared")
+        benchmark_duration = self.params.get("benchmark_duration", default=0)
+        transaction_count = self.params.get("transaction_count", default=0)
+
+        # validation: pgbench cannot accept both transaction count and duration
+        if benchmark_duration > 0 and transaction_count > 0:
+            self.log.warn("Pgbench cannot accept both max transaction count and duration of the test, proceeding with specified time limit")
+        elif benchmark_duration == 0 and transaction_count == 0:
+            self.log.warn("Transaction count / duration of the test not specified, defaulting to time limit of 120 secs")
+            benchmark_duration = 120
+
+        # create the db
+        self.run_cmd("createdb pgbench", False)
+
+        # setup tables
+        self.run_cmd(f"pgbench -i -s {scaling_factor} -n pgbench", False)
```

You shouldnt ignore errors for these command runs

---

yes I am specifying `ignore_failure` as False. So if there is an error , run_cmd() will just cancel the test. 
we can't proceed from this point if there is an error in this command